src/lucky/cookies/cookie_jar.cr

```diff
@@ -143,41 +142,21 @@ class Lucky::CookieJar
     end
   end
 
-  private def decrypt(cookie_value : String, cookie_name : String) : String
-    if encrypted_with_lucky?(cookie_value)
-      base_64_encrypted_part = cookie_value.lchop(LUCKY_ENCRYPTION_PREFIX)
-      decoded = Base64.decode(base_64_encrypted_part)
-      String.new(encryptor.decrypt(decoded))
-    elsif encrypted_with_legacy?(cookie_value)
-      decrypt(update_from_legacy_value(cookie_value), cookie_name)
-    else
-      raise <<-ERROR
-      It looks like this cookie's value is not encrypted by Lucky. This likely means the cookie was set by something other than Lucky, like a client side script.
-
-      You can access the raw value by using 'get_raw':
-
-          cookies.get_raw(:#{cookie_name}).value
-
-      ERROR
-    end
-  end
+  private def decrypt(cookie_value : String, cookie_name : String) : String?
+    return unless encrypted_with_lucky?(cookie_value)
 
-  private def update_from_legacy_value(value : String) : String
-    decoded_value = URI.decode_www_form(value)
-    LUCKY_ENCRYPTION_PREFIX + decoded_value.lchop(LEGACY_LUCKY_ENCRYPTION_PREFIX)
+    base_64_encrypted_part = cookie_value.lchop(LUCKY_ENCRYPTION_PREFIX)
+    decoded = Base64.decode(base_64_encrypted_part)
+    String.new(encryptor.decrypt(decoded))
+  rescue e
+    # an error happened while decrypting the cookie
+    # we will treat that as if no cookie was passed
```

Comment:
I think the reason they're only rescuing specific errors is that they don't want the blanket catch-all. If something goes wrong in an adjacent library which _isn't_ a decryption failure, it would cause a false positive on this rescue. 

The risk is, some code changes in Base64 or lchop or something which raises an exception, the tests pass for some reason, and now no cookies are readable and the code is just assuming it's a crypto failure.

Reply:
We could add a log here if we need. 

```
Lucky::Log.dexter.info { {message: "Cookie decrypt failed", error: e.message} }
```